Type checks for pd.options.display.* are not accurate

**Describe the bug**
Various display options underneath `pd.options.display.` do not have correct annotations per pandas documentation.

The following columns show mismatches between [documented behavior](https://pandas.pydata.org/docs/reference/api/pandas.describe_option.html#pandas.describe_option) compared to type check:
```
pd.describe_option('display.(?:colheader_justify|large_repr|max_columns|max_colwidth|max_dir_items|max_rows|max_seq_items|memory_usage|min_rows|show_dimensions)')
```

**To Reproduce**
```python
import pandas as pd

# Overly restrictive types set, expect errors to not be raised
pd.options.display.max_columns = None
pd.options.display.max_colwidth = None
pd.options.display.max_dir_items = None
pd.options.display.max_rows = None
pd.options.display.max_seq_items = None
pd.options.display.min_rows = None

# Overly permissive values permitted, expect errors to be raised
pd.options.display.colheader_justify = "invalid_value"
pd.options.display.large_repr = "invalid_value"
pd.options.display.memory_usage = "invalid_value"
pd.options.display.show_dimensions = "invalid_value"
```

```
test.py:4: error: Incompatible types in assignment (expression has type "None", variable has type "int")  [assignment]
    pd.options.display.max_columns = None
                                     ^~~~
test.py:5: error: Incompatible types in assignment (expression has type "None", variable has type "int")  [assignment]
    pd.options.display.max_colwidth = None
                                      ^~~~
test.py:6: error: Incompatible types in assignment (expression has type "None", variable has type "int")  [assignment]
    pd.options.display.max_dir_items = None
                                       ^~~~
test.py:7: error: Incompatible types in assignment (expression has type "None", variable has type "int")  [assignment]
    pd.options.display.max_rows = None
                                  ^~~~
test.py:8: error: Incompatible types in assignment (expression has type "None", variable has type "int")  [assignment]
    pd.options.display.max_seq_items = None
                                       ^~~~
test.py:9: error: Incompatible types in assignment (expression has type "None", variable has type "int")  [assignment]
    pd.options.display.min_rows = None
                                  ^~~~
test.py:14: error: Incompatible types in assignment (expression has type "str", variable has type "bool")  [assignment]
    pd.options.display.memory_usage = "invalid_value"
                                      ^~~~~~~~~~~~~~~
Found 7 errors in 1 file (checked 1 source file)
```


**Please complete the following information:**
 - OS: MacOS
 - OS Version: 15.5
 - python version: 3.12
 - pandas version: 2.2.3
 - version of type checker: mypy 1.9.0
 - version of installed `pandas-stubs`: 2.2.3.250308


**Additional context**
I'm happy to do PR, just let me know if that's preferred

--- a/pandas-stubs/_config/config.pyi
+++ b/pandas-stubs/_config/config.pyi
@@ -48,7 +48,7 @@ class DisplayUnicode(DictWrapper):
     east_asian_width: bool
 
 class Display(DictWrapper):
-    chop_threshold: int | None
+    chop_threshold: float | None
     colheader_justify: Literal["left", "right"]
     date_dayfirst: bool
     date_yearfirst: bool

--- a/pandas-stubs/_config/config.pyi
+++ b/pandas-stubs/_config/config.pyi
@@ -49,30 +49,30 @@ class DisplayUnicode(DictWrapper):
 
 class Display(DictWrapper):
     chop_threshold: int | None
-    colheader_justify: str
+    colheader_justify: Literal["left", "right"]
     date_dayfirst: bool
     date_yearfirst: bool
     encoding: str
     expand_frame_repr: bool
     float_format: Callable[[float], str] | None
     html: DisplayHTML
-    large_repr: str
+    large_repr: Literal["truncate", "info"]
     latex: DisplayLaTeX
     max_categories: int
-    max_columns: int
-    max_colwidth: int
-    max_dir_items: int
+    max_columns: int | None
+    max_colwidth: int | None
+    max_dir_items: int | None
     max_info_columns: int
     max_info_rows: int
-    max_rows: int
-    max_seq_items: int
-    memory_usage: bool
-    min_rows: int
+    max_rows: int | None
+    max_seq_items: int | None
+    memory_usage: bool | Literal["deep"] | None
+    min_rows: int | None
     multi_sparse: bool
     notebook_repr_html: bool
     pprint_nest_depth: int
     precision: int
-    show_dimensions: str
+    show_dimensions: bool | Literal["truncate"]
     unicode: DisplayUnicode
     width: int
 

--- a/tests/test_config.py
+++ b/tests/test_config.py
@@ -54,13 +54,13 @@ def test_display_float_format():
 def test_display_types_none_allowed_get_options():
     # GH 1230
     # Initial values
-    assert_type(pd.options.display.chop_threshold, Optional[float])
-    assert_type(pd.options.display.max_columns, Optional[int])
-    assert_type(pd.options.display.max_colwidth, Optional[int])
-    assert_type(pd.options.display.max_dir_items, Optional[int])
-    assert_type(pd.options.display.max_rows, Optional[int])
-    assert_type(pd.options.display.max_seq_items, Optional[int])
-    assert_type(pd.options.display.min_rows, Optional[int])
+    check(assert_type(pd.options.display.chop_threshold, Optional[float]), type(None))
+    check(assert_type(pd.options.display.max_columns, Optional[int]), int)
+    check(assert_type(pd.options.display.max_colwidth, Optional[int]), int)
+    check(assert_type(pd.options.display.max_dir_items, Optional[int]), int)
+    check(assert_type(pd.options.display.max_rows, Optional[int]), int)
+    check(assert_type(pd.options.display.max_seq_items, Optional[int]), int)
+    check(assert_type(pd.options.display.min_rows, Optional[int]), int)
 
 
 def test_display_types_none_allowed_set_options():

--- a/tests/test_config.py
+++ b/tests/test_config.py
@@ -2,6 +2,7 @@ from typing import (
     TYPE_CHECKING,
     Any,
     Callable,
+    Literal,
     Optional,
 )
 
@@ -48,3 +49,78 @@ def test_display_float_format():
     formatter = "{,.2f}".format
     with pd.option_context("display.float_format", formatter):
         assert pd.get_option("display.float_format") == formatter
+
+
+def test_display_types_none_allowed():
+    # GH 1230
+    # Initial values
+    assert_type(pd.options.display.max_columns, Optional[int])
+    assert_type(pd.options.display.max_colwidth, Optional[int])
+    assert_type(pd.options.display.max_dir_items, Optional[int])
+    assert_type(pd.options.display.max_rows, Optional[int])
+    assert_type(pd.options.display.max_seq_items, Optional[int])
+    assert_type(pd.options.display.min_rows, Optional[int])
+    # Test with None
+    pd.options.display.max_columns = None
+    check(assert_type(pd.options.display.max_columns, None), type(None))
+    pd.options.display.max_colwidth = None
+    check(assert_type(pd.options.display.max_colwidth, None), type(None))
+    pd.options.display.max_dir_items = None
+    check(assert_type(pd.options.display.max_dir_items, None), type(None))
+    pd.options.display.max_rows = None
+    check(assert_type(pd.options.display.max_rows, None), type(None))
+    pd.options.display.max_seq_items = None
+    check(assert_type(pd.options.display.max_seq_items, None), type(None))
+    pd.options.display.min_rows = None
+    check(assert_type(pd.options.display.min_rows, None), type(None))
+    # Test with integer values
+    pd.options.display.max_columns = 100
+    check(assert_type(pd.options.display.max_columns, int), int)
+    pd.options.display.max_colwidth = 100
+    check(assert_type(pd.options.display.max_colwidth, int), int)
+    pd.options.display.max_dir_items = 100
+    check(assert_type(pd.options.display.max_dir_items, int), int)
+    pd.options.display.max_rows = 100
+    check(assert_type(pd.options.display.max_rows, int), int)
+    pd.options.display.max_seq_items = 100
+    check(assert_type(pd.options.display.max_seq_items, int), int)
+    pd.options.display.min_rows = 100
+    check(assert_type(pd.options.display.min_rows, int), int)
+
+
+def test_display_types_literal_constraints():
+    # GH 1230
+    # Various display options have specific allowed values
+    # Test colheader_justify with allowed values
+    assert_type(pd.options.display.colheader_justify, Literal["left", "right"])
+    pd.options.display.colheader_justify = "left"
+    check(assert_type(pd.options.display.colheader_justify, Literal["left"]), str)
+    pd.options.display.colheader_justify = "right"
+    check(assert_type(pd.options.display.colheader_justify, Literal["right"]), str)
+
+    # Test large_repr with allowed values
+    assert_type(pd.options.display.large_repr, Literal["truncate", "info"])
+    pd.options.display.large_repr = "truncate"
+    check(assert_type(pd.options.display.large_repr, Literal["truncate"]), str)
+    pd.options.display.large_repr = "info"
+    check(assert_type(pd.options.display.large_repr, Literal["info"]), str)
+
+    # Test memory_usage with allowed values
+    assert_type(pd.options.display.memory_usage, Optional[Literal[True, False, "deep"]])
+    pd.options.display.memory_usage = True
+    check(assert_type(pd.options.display.memory_usage, Literal[True]), bool)
+    pd.options.display.memory_usage = False
+    check(assert_type(pd.options.display.memory_usage, Literal[False]), bool)
+    pd.options.display.memory_usage = "deep"
+    check(assert_type(pd.options.display.memory_usage, Literal["deep"]), str)
+    pd.options.display.memory_usage = None
+    check(assert_type(pd.options.display.memory_usage, None), type(None))
+
+    # Test show_dimensions with allowed values
+    assert_type(pd.options.display.show_dimensions, Literal[True, False, "truncate"])
+    pd.options.display.show_dimensions = True
+    check(assert_type(pd.options.display.show_dimensions, Literal[True]), bool)
+    pd.options.display.show_dimensions = False
+    check(assert_type(pd.options.display.show_dimensions, Literal[False]), bool)
+    pd.options.display.show_dimensions = "truncate"
+    check(assert_type(pd.options.display.show_dimensions, Literal["truncate"]), str)

--- a/tests/test_config.py
+++ b/tests/test_config.py
@@ -51,41 +51,35 @@ def test_display_float_format():
         assert pd.get_option("display.float_format") == formatter
 
 
-def test_display_types_none_allowed():
+def test_display_types_none_allowed_get_options():
     # GH 1230
     # Initial values
+    assert_type(pd.options.display.chop_threshold, Optional[float])
     assert_type(pd.options.display.max_columns, Optional[int])
     assert_type(pd.options.display.max_colwidth, Optional[int])
     assert_type(pd.options.display.max_dir_items, Optional[int])
     assert_type(pd.options.display.max_rows, Optional[int])
     assert_type(pd.options.display.max_seq_items, Optional[int])
     assert_type(pd.options.display.min_rows, Optional[int])
-    # Test with None
+
+
+def test_display_types_none_allowed_set_options():
+    # GH 1230
+    # Test setting each option as None and then to a specific value
+    pd.options.display.chop_threshold = None
+    pd.options.display.chop_threshold = 0.9
     pd.options.display.max_columns = None
-    check(assert_type(pd.options.display.max_columns, None), type(None))
-    pd.options.display.max_colwidth = None
-    check(assert_type(pd.options.display.max_colwidth, None), type(None))
-    pd.options.display.max_dir_items = None
-    check(assert_type(pd.options.display.max_dir_items, None), type(None))
-    pd.options.display.max_rows = None
-    check(assert_type(pd.options.display.max_rows, None), type(None))
-    pd.options.display.max_seq_items = None
-    check(assert_type(pd.options.display.max_seq_items, None), type(None))
-    pd.options.display.min_rows = None
-    check(assert_type(pd.options.display.min_rows, None), type(None))
-    # Test with integer values
     pd.options.display.max_columns = 100
-    check(assert_type(pd.options.display.max_columns, int), int)
+    pd.options.display.max_colwidth = None
     pd.options.display.max_colwidth = 100
-    check(assert_type(pd.options.display.max_colwidth, int), int)
+    pd.options.display.max_dir_items = None
     pd.options.display.max_dir_items = 100
-    check(assert_type(pd.options.display.max_dir_items, int), int)
+    pd.options.display.max_rows = None
     pd.options.display.max_rows = 100
-    check(assert_type(pd.options.display.max_rows, int), int)
+    pd.options.display.max_seq_items = None
     pd.options.display.max_seq_items = 100
-    check(assert_type(pd.options.display.max_seq_items, int), int)
+    pd.options.display.min_rows = None
     pd.options.display.min_rows = 100
-    check(assert_type(pd.options.display.min_rows, int), int)
 
 
 def test_display_types_literal_constraints():

